.github/workflows/ci_tests.yaml

```diff
@@ -0,0 +1,99 @@
+        with:
+          # fecth all history so that versioneer works
+          fetch-depth: 0
+
+      # Setup Miniconda
+      - name: Setup Miniconda
+        uses: goanpeca/setup-miniconda@v1.6.0
+        with:
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install GMT and required dependencies
+        shell: bash -l {0}
+        run: |
+          requirements_file=full-conda-requirements.txt
+          cat requirements.txt requirements-dev.txt > $requirements_file
+          cat << EOF >> $requirements_file
+          gmt=6.0.0
+          make
+          EOF
+          conda install --yes --file $requirements_file
+
+      # Show installed pkg information for postmortem diagnostic
+      - name: List installed packages
+        shell: bash -l {0}
+        run: conda list
+
+      # Cache the ${HOME}/.gmt directory, for docs and testing
+      - name: Cache GMT directory
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            ~/.gmt/cache
+            ~/.gmt/server
+          key: cache-gmt-${{ github.ref }}-${{ runner.os }}-20200609
+          restore-keys: cache-gmt-refs/heads/master-
+
+      # Workaround for the timeouts of 'gmt which' on Linux and Windows
+      - name: Download remote data using wget (Linux & Windows)
+        shell: bash -l {0}
+        run: |
+          if [ "$RUNNER_OS" == "Windows" ]; then choco install wget; fi  # install wget on Windows
+          mkdir ~/.gmt ~/.gmt/cache ~/.gmt/server
+          wget --no-check-certificate https://oceania.generic-mapping-tools.org/gmt_hash_server.txt -P ~/.gmt/server/
+          for data in earth_relief_01d.grd earth_relief_30m.grd earth_relief_10m.grd; do
+            wget --no-check-certificate https://oceania.generic-mapping-tools.org/${data} -P ~/.gmt/server/
+          done
+          for data in ridge.txt Table_5_11.txt tut_bathy.nc tut_quakes.ngdc tut_ship.xyz usgs_quakes_22.txt; do
+            wget --no-check-certificate https://oceania.generic-mapping-tools.org/cache/${data} -P ~/.gmt/cache/
+          done
+        if: steps.cache.outputs.cache-hit != 'true' && runner.os != 'macOS'
+
+      # Download remote files, if not already cached
+      - name: Download remote data (macOS)
+        shell: bash -l {0}
+        run: gmt which -Gu @earth_relief_01d @earth_relief_30m @earth_relief_10m @ridge.txt @Table_5_11.txt @tut_bathy.nc @tut_quakes.ngdc @tut_ship.xyz @usgs_quakes_22.txt
+        if: steps.cache.outputs.cache-hit != 'true' && runner.os == 'macOS'
+
+      # Install the package that we want to test
+      - name: Install the package
+        shell: bash -l {0}
+        run: |
+          python setup.py sdist --formats=zip
+          pip install dist/*
+
+      # Run the tests
+      - name: Test with pytest
+        shell: bash -l {0}
+        run: make test PYTEST_EXTRA="-r P"
```

Comment:
Is the `PYTEST_EXTRA` necessary? I never understand what does it do.

Reply:
On our travis.yml, it says https://github.com/GenericMappingTools/pygmt/blob/6b8d159454b07f3694117da530441835352f6504/.travis.yml#L65-L66

Might be best to leave it so that stdout and stderr is captured.

Reply:
The variable is not used in any files of pygmt. I also can't find it in pytest documentation, so not an pytest environmental variable. Confusing.

Reply:
I can trace it back using `git blame` to #188. But yeah, can't find it in the pytest docs either, it's strange.

Reply:
Ok, found it in the commandline.

```
-r chars              show extra test summary info as specified by chars: (f)ailed, (E)rror, (s)kipped, (x)failed, (X)passed, (p)assed, (P)assed with output, (a)ll except passed
                        (p/P), or (A)ll. (w)arnings are enabled by default (see --disable-warnings), 'N' can be used to reset the list. (default: 'fE').
```

Reply:
I just ran `make test` and `make test PYTEST_EXTRA="-r P"`, the output are exactly the same, except the random directory names for images. But if I add `-r P` to the variable `PYTEST_ARGS` in Makefile, I can see more output like:
```
> basemap [WARNING]: Your primary y annotation interval exceeds the secondary interval.
> basemap [WARNING]: GMT expects it to be the other way around (primary annotations are closest to axis, secondary are further away)
> basemap [WARNING]: Consider correcting your command - proceeding with your selections
> basemap [WARNING]: Your primary y tick interval exceeds the secondary interval.
> basemap [WARNING]: GMT expects it to be the other way around (primary annotations are closest to axis, secondary are further away)
> basemap [WARNING]: Consider correcting your command - proceeding with your selections
> basemap [WARNING]: Your primary y annotation interval exceeds the secondary interval.
> basemap [WARNING]: GMT expects it to be the other way around (primary annotations are closest to axis, secondary are further away)
> basemap [WARNING]: Consider correcting your command - proceeding with your selections
> basemap [WARNING]: Your primary y tick interval exceeds the secondary interval.
> basemap [WARNING]: GMT expects it to be the other way around (primary annotations are closest to axis, secondary are further away)
> basemap [WARNING]: Consider correcting your command - proceeding with your selections
```

Reply:
The extra outputs are useful. I think we should change the Makefile instead. 

Changing 
```
PYTEST_ARGS=--cov-config=../.coveragerc --cov-report=term-missing --cov=$(PROJECT) --doctest-modules -v --mpl --mpl-results-path=results --pyargs
```
to 
```
PYTEST_ARGS=--cov-config=../.coveragerc --cov-report=term-missing --cov=$(PROJECT) --doctest-modules -v --mpl --mpl-results-path=results --pyargs ${PYTEST_EXTRA}
```